### PR TITLE
Polish

### PR DESCRIPTION
**Pull request by mikez321**
What started as polish up ended up being a massive overhaul.
Removes all instance variables and initialize from enigma class.
Creates a new dedicated class for date and key verification.
Moves all verification tests to verification test class.
Edits all classes to work with verified and randomly created data.
All tests pass rake, simplecov at 100%, pry interaction passes, and CLI works.

### CODE REVIEW

File: lib/decrypt.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 require './lib/enigma'
 encrypted_message = File.open(ARGV[0]).read.chomp
 enigma = Enigma.new
+
 decoded_message = enigma.decrypt(encrypted_message, ARGV[2], ARGV[3])
 File.write(ARGV[1], decoded_message[:decryption])
-puts "Created #{ARGV[1]} with the key #{ARGV[2]} and date #{ARGV[3]}"
+puts "Created #{ARGV[1]} with the key #{decoded_message[:key]} and date #{decoded_message[:date]}"
```
Comment (mikez321) [inline lib/decrypt.rb:7]: Nice job pulling from this information you already had on hand.

File: lib/shifter.rb [hunk 2]
```diff
@@ -1,22 +1,37 @@
+require './lib/validator'
 require './modules/referenceable'
 class Shifter
   include Referenceable
-  attr_reader :key, :date
-  def initialize(key, date)
-    @key = key
-    @date = date
+  attr_reader :key_to_use, :date_to_use
+  def initialize(passed_key = nil, passed_date = nil)
```
Comment (mikez321) [inline lib/shifter.rb:6]: Not sure if all these are totally needed, but as long as they pass.

File: lib/validator.rb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+  def validate_start_date
+    if valid_start_date?
+      @valid_date = @start_date
+    else
+      @valid_date = Date.today.strftime("%d%m%y")
+    end
+  end
+
+  # def error_message
```
Comment (mikez321) [inline lib/validator.rb]: This needs to be removed before submission.

File: test/enigma_test.rb [hunk 4]
```diff
@@ -9,73 +9,27 @@
 class EnigmaTest < Minitest::Test
 
   def test_it_exists
-    enigma = Enigma.new("02715", "040895")
```
Comment (mikez321) [inline test/enigma_test.rb:12]: That's a lot of stuff being pulled out.  Nice work.

File: test/validator_test.rb [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    validator1 = Validator.new("02715")
+    assert_equal true, validator1.start_key_only_numbers?
+    validator2 = Validator.new("ABCD")
+    assert_equal false, validator2.start_key_only_numbers?
+    validator3 = Validator.new("AB.89!")
+    assert_equal false, validator3.start_key_only_numbers?
+
+  end
+
+  def test_it_can_test_for_a_valid_key
```
Comment (mikez321) [inline test/validator_test.rb:40]: Nice work on the edge cases.

File: test/validator_test.rb [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    validator3 = Validator.new("Nick")
+    assert_equal false, validator3.valid_start_key?
+
+    validator4 = Validator.new("1")
+    assert_equal false, validator4.valid_start_key?
+  end
+
+  def test_if_an_incorrect_key_is_assigned_it_will_assign_a_random_key
+    validator1 = Validator.new("ABCD")
+    validator1.stubs(:rand).returns("3907")
```
Comment (mikez321) [inline test/validator_test.rb:56]: In case it wasn't mentioned before, good job using a stub here.

**Review by mikez321 — COMMENTED** [review]
Remove those comments and you'll be good to go!

**Comment by mikez321** [comment]
Thanks for removing those commented lines.  Great work on this project!
